### PR TITLE
[WFLY-16471] Update http-custom-mechanism quickstart

### PR DESCRIPTION
**Pull request by soul2zimate**
Issue: https://issues.redhat.com/browse/WFLY-16471

1. remove unneeded `configure-security-domain.cli`.
2. fix parent for `custom-module/pom.xml` and rename its `artifactId` to `custom-module`. (not duplicate with parent `artifactId` in `http-custom-mechanism/pom.xml` (update accordingly in `add-custom-module.cli` and `remove-custom-module.cli`)
3. add `restore-configuration.cli`

@fjuma  please review.

### CODE REVIEW

File: http-custom-mechanism/custom-module/add-custom-module.cli [hunk 1]
```diff
@@ -1,3 +1,3 @@
 # Add the custom module to the server
 
-module add --name=org.jboss.as.quickstart.http_custom_mechanism.custom-http-mechanism --resources=target/http-custom-mechanism.jar --dependencies=org.wildfly.security.elytron,javax.api
+module add --name=org.jboss.as.quickstart.http_custom_mechanism.custom-http-mechanism --resources=target/custom-module.jar --dependencies=org.wildfly.security.elytron,javax.api
```
Comment (fjuma) [inline http-custom-mechanism/custom-module/add-custom-module.cli:3]: Why is this being renamed?

File: http-custom-mechanism/custom-module/pom.xml [hunk 2]
```diff
@@ -20,16 +20,12 @@
...
-        -->
-        <version>2</version>
-        <relativePath/>
+        <artifactId>http-custom-mechanism</artifactId>
+        <version>27.0.0.Beta1-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
     </parent>
 
-    <artifactId>http-custom-mechanism</artifactId>
+    <artifactId>custom-module</artifactId>
```
Comment (fjuma) [inline http-custom-mechanism/custom-module/pom.xml:28]: Why is this being renamed?

File: http-custom-mechanism/custom-module/add-custom-module.cli (hunk 1 above)
Comment (fjuma) [reply http-custom-mechanism/custom-module/add-custom-module.cli:3]: Actually, just noticed that you already mentioned this in the description so you can ignore this comment :)

File: http-custom-mechanism/custom-module/pom.xml (hunk 2 above)
Comment (fjuma) [reply http-custom-mechanism/custom-module/pom.xml:28]: Actually, just noticed that you already mentioned this in the description so you can ignore this comment :)

File: http-custom-mechanism/README.adoc [hunk 3]
```diff
@@ -271,8 +261,20 @@ Current Principal 'quickstartUser'    </p>
 // Undeploy the Quickstart
 include::../shared-doc/undeploy-the-quickstart.adoc[leveloffset=+1]
 // Restore the {productName} Standalone Server Configuration Manually
+:restoreScriptName: restore-configuration.cli
 include::../shared-doc/restore-standalone-server-configuration-manual.adoc[leveloffset=+2]
 
+// Additional information about this script
+This script reverts the changes made to the `undertow`  and `elytron` subsystem.You should see the following result when you run the script.
```
Comment (fjuma) [inline http-custom-mechanism/README.adoc:268]: s/subsystem.You/subsystems. You

**Review by fjuma — COMMENTED** [review]
Thanks for working on this, @soul2zimate! Just added a small comment.

File: http-custom-mechanism/README.adoc (hunk 3 above)
Comment (fjuma) [reply http-custom-mechanism/README.adoc:268]: Another typo I noticed outside of your changes that would be good to correct here is in the "Configure the Server to Use the Custom Module" section:

"This script adds the configuration that enables Elytron security fto use" ->
"This script adds the configuration that enables Elytron security to use"

Comment (soul2zimate) [reply http-custom-mechanism/README.adoc]: Thanks, I have fixed the spellings

**Pull request merged**